src/cpp/fastdds/xtypes/type_representation/TypeObjectRegistry.hpp

```diff
@@ -46,6 +46,8 @@ struct hash<eprosima::fastdds::dds::xtypes::TypeIdentifier>
 
 };
 
+#ifndef TYPE_IDENTIFIER_WITH_SIZE_HASH_SPECIALIZATION
```

Comment:
Define this hash in a common header.

Reply:
Created header in [ee520e4](https://github.com/eProsima/Fast-DDS/pull/4177/commits/ee520e48f15c205983ef93344d678cca12d35d0b)